lib/bolt/inventory/group.rb

```diff
@@ -68,7 +68,7 @@ def initialize(input, plugins)
         # though, since that will resolve any nested references and we want to
         # leave it to the group to do that lazily.
         groups = @plugins.resolve_top_level_references(groups)
-
+        raise ValidationError.new("Groups must be an Array", nil) unless groups.is_a?(Array)
```

Comment:
Interesting. The call to `Array(groups)` below this would seem to indicate this doesn't _have_ to be an array. I think the problem is actually that passing a hash to `Array()` actually converts the hash to an array of key/value pairs rather than wrapping it in an array. So maybe `Array(groups)` should be `groups = [groups] unless groups.is_a?(Array)`.

Reply:
I was under the impression that `Array(...)` _always_ wrapped the argument in an array if it wasn't one, but I guess that's not the case. I wonder how many other places in Bolt this same bug arises from the same incorrect assumption.

Reply:
It appears there will be a similar bug for individual targets. https://github.com/puppetlabs/bolt/pull/2416/files#diff-297edc81f841f0d72ea66094838373e2eae905ddfcf1335e5f020411b3e615deR48

Reply:
Yeah, this is a bigger issue than I originally expected. Ticketed for Bolt team https://github.com/puppetlabs/bolt/issues/2430